packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1302,11 +2003,24 @@ class RoundSliderThumbShape extends SliderComponentShape {
       begin: sliderTheme.disabledThumbColor,
       end: sliderTheme.thumbColor,
     );
+
+    final Size size = getPreferredSize(true, isDiscrete);
+    // Add a stroke of 1dp around the circle if this thumb would overlap
+    // the other thumb.
+    if (isOnTop) {
+      canvas.drawCircle(
+        center,
+        (size.width + 2) / 2,
+        Paint()..color = sliderTheme.thumbStrokeColor,
```

Comment:
Usually we try to stick to theming variables being overridable on the individual widgets, should we consider adding `Color thumbStrokeColor` as a field on this class, then doing `thumbStrokeColor ?? sliderTheme.thumbStrokeColor` here?

Reply:
Slider is a bit different on the way it handles this. This slider theme these painters get is actually a copy where it already resolved any widget specific inputs (see line 511 in range_slider.dart)